### PR TITLE
Add rust-based crypto engine

### PR DESCRIPTION
**Pull request by SantiagoTorres**
I've found it that cpabe is just impossible to make portably. Three depedencies
built on top of each other and symbols being not-exported makes Petra very hard
to use. The current implementation basically popens into cpabe-*, which relies
on the filesystem to actually work. I don't think this is a good idea.

Instead, we add a rust-based portable python binding using pyo3 and rabe. This
allows us to have more control over the data and messaging. Rust should be
equally efficient, though encoding messages in json back and forth may incur in
some cost.

### CODE REVIEW

File: privateSBOMExchange/src/cpabe/test.py
Comment (marcelamelara) [inline privateSBOMExchange/src/cpabe/test.py:1]: Can we move this file into a dedicated `/tests` directory?

File: privateSBOMExchange/src/cpabe/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        attr_vec.push(value.to_string());
+    }
+    
+    let pos: Vec<&str> = attr_vec.iter().map(AsRef::as_ref).collect();
+    let sk = delegate(&_pk, &_sk, pos.as_slice()).unwrap();
+    sk_to_json(sk)
+}
+
+
+fn pk_to_json(a: CpAbePublicKey) -> String {
```
Comment (marcelamelara) [inline privateSBOMExchange/src/cpabe/src/lib.rs:71]: I'm assuming this means keys won't be in PEM format? We might want to document this.

**Review by marcelamelara — COMMENTED** [review]
Thanks @SantiagoTorres ! This is way more portable! I do think we should start moving tests over to a dedicated directory in this PR.

File: privateSBOMExchange/src/cpabe/test.py
Comment (marcelamelara) [reply privateSBOMExchange/src/cpabe/test.py:1]: Ah, I didn't realize we have #19 open. Let's resolve this as part of that PR.

File: privateSBOMExchange/src/cpabe/src/lib.rs (hunk 1 above)
Comment (SantiagoTorres) [reply privateSBOMExchange/src/cpabe/src/lib.rs:71]: yeah, I don't think that cp-abe keys have a supported PEM format without shoehorning them in another format. I'll add comments in a new commit

Comment (marcelamelara) [reply privateSBOMExchange/src/cpabe/src/lib.rs:71]: Thanks! Yeah, those comments will be a helpful note to self that you can't just import keys etc.

**Review by marcelamelara — APPROVED** [review]
Thanks for the updates @SantiagoTorres !!

**Review by abuishgair — APPROVED** [review]

**Pull request merged**